days/day06/solve.py

```diff
@@ -0,0 +1,29 @@
+with open("input.txt") as input_file:
+    input = input_file.read().strip()
+
+START_PACKET_MARKER_LENGTH = 4
+START_MESSAGE_MARKER_LENGTH = 14
+
+
+def find_first_marker_index(letters, marker_length):
+    queue = []
+    result = None
+
+    for index, letter in enumerate(letters):
+        if len(queue) < marker_length:
+            queue.append(letter)
+            continue
+        if len(set(queue)) == marker_length:
+            result = index
+            break
+        queue.pop(0)
+        queue.append(letter)
+
+    return result
```

Comment:
As you told me, could have used slices to do that
Just an example of solution using slices:
```python
def find_first_marker_index_1(letters: str, marker_length: int) -> int:
    marker_index = -1
    for i in range(len(letters) - marker_length + 1):
        if len(set(letters[i : i + marker_length])) == marker_length:
            marker_index = i
            break

    return marker_index
```